src/gui/Game.tsx

```diff
@@ -36,6 +36,8 @@ export class Game extends React.Component<{ gameId: number, name: string, isRepl
 
   private viewerState: ViewerState
 
+  private status: MessageContent.StatusReportContent
```

Comment:
Das scheint nirgends benutzt zu werden.

Reply:
Strg+F -> `this.status`

sicher?

Reply:
Okay, hatte das in der render-Methode nicht gesehen. Hast du darueber nachgedacht, dass in den State der Komponente zu tun? Was spricht dagegen?

Reply:
Habe ich, ich wollte es sogar tun, aber das erzeugt folgendes Problem: 
`render` ruft `updateViewer` auf. Sobald die Promise in `updateViewer` resolved wird, wird durch `setState` der status geupdated und damit erneut `render` aufgerufen.
Damit kann man ziemlich schnell einen computer zum erliegen bekommen ^^

Reply:
Sowas dachte ich mir schon. Trotzdem gefaellt es mir nicht, dass wir hier an dem Konzept von React vorbeiprogrammieren.

Siehe https://stackoverflow.com/questions/24718709/reactjs-does-render-get-called-any-time-setstate-is-called/24719289#24719289

Wir koennen beeinflussen, ob render aufgerufen wird. Fuer diese Komponente koennte man `shouldComponentUpdate` so abaendern, dass render nur bei einer Aenderung des State aufgerufen wird. Das muesste das Problem loesen. Was denkst du?

Reply:
Mein Überlegung war eher, ob updateViewer nicht separat dann gecallt werden sollte, wenn es wirklich nötig ist, und nicht grundsätzlich bei jedem render.

Reply:
updateViewer uebertraegt ja auch den Zustand zum eigentlichen Viewer hin (der keine React-Komponente ist). Daher muss es aufgerufen werden, wenn sich der State der Game-Komponente aendert (also z.B. die aktuelle Runde oder sogar das aktuell anzuzeigende Spiel). Wir haben da eine Kreis-Abhaengigkeit, fuerchte ich.